### DISCUSSION TITLE
How unique do IDs have to be in Lowdefy?

### DISCUSSION THREAD

**Discussion by vrilcode** [lowdefy/lowdefy#1466] [Questions and Help] [answered]
According to docs, blocks and other concepts in Lowdefy need IDs, which must be unique. I found no information about the "scope of uniqueness". How unique should block IDs, request IDs, action IDs, … be? Unique per page, per app, per …?
Maybe I have overseen parts of the docs. Would be glad, if someone can enlighten me. 💡

**Comment by SamTolmay** [comment, marked as answer]
Hi @vrilcode 

ID's in Lowdefy only need to be unique in the "scope in which they operate". Specifically:
- Page IDs should be unique across the entire app. The pageId determines the route under which the app is served.
- Request IDs should be unique per page.
- In version 4, block IDs no longer need to be unique. This often makes references of shared components simpler. If you have more than 1 input block with the same ID, they will share a value. If you use CallMethod  with a non-unique
- Action IDs need to be unique in the action chain which they execute in, so you can have duplicate ids for different blocks on the same page.

I will review the docs and make sure this is a little clearer.

**Reply by vrilcode** [reply]
Thank you!
